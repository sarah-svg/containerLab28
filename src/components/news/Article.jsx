import React from "react";
import PropTypes from "prop-types";

const Article = ({  title, author, description }) => {
  return (
    <>
      <a  style={{ fontSize: "1.2rem" }}>
        {title}
      </a>
      {author && <p>Written by: {author}</p>}
      <p>{description}</p>
    </>
  );
};

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  description: PropTypes.string,
};

export default Article;
