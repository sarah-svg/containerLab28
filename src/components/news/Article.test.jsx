import React from "react";
import { render, cleanup } from "@testing-library/react";
import Article from "./Article.jsx";

describe("Article component", () => {
  afterEach(() => cleanup());
  it("renders Article", () => {
    const { asFragment } = render(<Article />);
    expect(asFragment()).toMatchSnapshot();
  });
});
