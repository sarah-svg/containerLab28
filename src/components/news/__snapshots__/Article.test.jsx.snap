// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <a
    style="font-size: 1.2rem;"
  />
  <p />
</DocumentFragment>
`;
